ftw/tabbedview/caching.py

```diff
@@ -0,0 +1,24 @@
+from ftw.dictstorage.interfaces import IDictStorage
+from ftw.tabbedview.interfaces import IDefaultTabStorageKeyGenerator
+from plone.app.caching.interfaces import IETagValue
+from plone.app.caching.operations.utils import getContext
+from zope.component import adapts
+from zope.component import getMultiAdapter
+from zope.interface import implements
+from zope.interface import Interface
+
+
+class TabbedviewETagValue(object):
+    implements(IETagValue)
+    adapts(Interface, Interface)
+
+    def __init__(self, published, request):
+        self.published = published
+        self.request = request
+
+    def __call__(self):
+        storage_key_generator = getMultiAdapter(
```

Comment:
uh no, no `isinstance` please, an `ITabbedViewView.providedBy()` would be ok though, should such an interface exist. but not necessary if it works for any view.